### PR TITLE
ath79: Add support for OpenMesh A60 family

### PR DESCRIPTION
**Pull request by ecsv**
This PR is based on the PR #3710 + #3712

All these devices were part of the A60 image in the ar71xx target. These were now converted to ath79 and split into different images in this process

OpenMesh A40
============

Device specifications:
----------------------

* Qualcomm/Atheros QCA9558 ver 1 rev 0
* 720/600/240 MHz (CPU/DDR/AHB)
* 128 MB of RAM
* 16 MB of SPI NOR flash
  - 2x 7 MB available; but one of the 7 MB regions is the recovery image
* 2T2R 2.4 GHz Wi-Fi (11n)
* 2T2R 5 GHz Wi-Fi (11ac)
* multi-color LED (controlled via red/green/blue GPIOs)
* 1x GPIO-button (reset)
* external h/w watchdog (enabled by default))
* TTL pins are on board (arrow points to VCC, then follows: GND, TX, RX)
* 2x ethernet
  - eth0
    + Label: Ethernet 1
    + 10/100/1000 Mbps Ethernet
    + 802.3af POE
    + used as WAN interface
  - eth1
    + Label: Ethernet 1
    + 10/100/1000 Mbps Ethernet
    + used as LAN interface
* 1x USB
* internal antennas

Flashing instructions:
----------------------

Various methods can be used to install the actual image on the flash.
Two easy ones are:

### ap51-flash

The tool ap51-flash (https://github.com/ap51-flash/ap51-flash) should be used to transfer the image to the u-boot when the device boots up.

### initramfs from TFTP

The serial console must be used to access the u-boot shell during bootup. It can then be used to first boot up the initramfs image from a TFTP server (here with the IP 192.168.1.21):

    setenv serverip 192.168.1.21
    setenv ipaddr 192.168.1.1
    tftpboot 0c00000 <filename-of-initramfs-kernel>.bin && bootm $fileaddr

The actual sysupgrade image can then be transferred (on the LAN port) to the device via

    scp <filename-of-squashfs-sysupgrade>.bin root@192.168.1.1:/tmp/

On the device, the sysupgrade must then be started using

    sysupgrade -n /tmp/<filename-of-squashfs-sysupgrade>.bin

OpenMesh A60
============

Device specifications:
----------------------

* Qualcomm/Atheros QCA9558 ver 1 rev 0
* 720/600/240 MHz (CPU/DDR/AHB)
* 128 MB of RAM
* 16 MB of SPI NOR flash
  - 2x 7 MB available; but one of the 7 MB regions is the recovery image
* 3T3R 2.4 GHz Wi-Fi (11n)
* 3T3R 5 GHz Wi-Fi (11ac)
* multi-color LED (controlled via red/green/blue GPIOs)
* 1x GPIO-button (reset)
* external h/w watchdog (enabled by default))
* TTL pins are on board (arrow points to VCC, then follows: GND, TX, RX)
* 2x ethernet
  - eth0
    + Label: Ethernet 1
    + 10/100/1000 Mbps Ethernet
    + 802.3af POE
    + used as WAN interface
  - eth1
    + Label: Ethernet 1
    + 10/100/1000 Mbps Ethernet
    + used as LAN interface
* 1x USB
* internal antennas

Flashing instructions:
----------------------

Various methods can be used to install the actual image on the flash.
Two easy ones are:

### ap51-flash

The tool ap51-flash (https://github.com/ap51-flash/ap51-flash) should be used to transfer the image to the u-boot when the device boots up.

### initramfs from TFTP

The serial console must be used to access the u-boot shell during bootup. It can then be used to first boot up the initramfs image from a TFTP server (here with the IP 192.168.1.21):

    setenv serverip 192.168.1.21
    setenv ipaddr 192.168.1.1
    tftpboot 0c00000 <filename-of-initramfs-kernel>.bin && bootm $fileaddr

The actual sysupgrade image can then be transferred (on the LAN port) to the device via

    scp <filename-of-squashfs-sysupgrade>.bin root@192.168.1.1:/tmp/

On the device, the sysupgrade must then be started using

    sysupgrade -n /tmp/<filename-of-squashfs-sysupgrade>.bin

### CODE REVIEW

File: target/linux/ath79/generic/base-files/lib/upgrade/openmesh.sh [hunk 1]
```diff
@@ -68,6 +68,7 @@ platform_do_upgrade_openmesh()
 	# boot anymore to Linux until it was reflashed with ap51-flash.
 	local next_boot_part="1"
 	case "$(board_name)" in
+	openmesh,a40|\
```
Comment (adschm) [inline target/linux/ath79/generic/base-files/lib/upgrade/openmesh.sh:71]: So, all currently known devices have the same value?  I'd even consider to hard-code it then, or follow the function argument idea from the om2p comment. Whatever gets us rid of the case in openmesh.sh

Comment (ecsv) [reply target/linux/ath79/generic/base-files/lib/upgrade/openmesh.sh:71]: consider this done (but I will push it after I've changed it in all PRs)

File: target/linux/ath79/dts/qca9558_openmesh_a60.dtsi [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+
+&usb1 {
+	#address-cells = <1>;
+	#size-cells = <0>;
+	status = "okay";
+
+	hub_port1: port@1 {
+		reg = <1>;
+		#trigger-source-cells = <0>;
+	};
```
Comment (adschm) [inline target/linux/ath79/dts/qca9558_openmesh_a60.dtsi:77]: port is only needed when a LED is linked to it. You may remove everything except the status=okay from the usb1 node here.

File: target/linux/ath79/dts/qca9558_openmesh_a60.dtsi [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+	mtd-mac-address = <&art 0x0>;
+
+	phy-mode = "rgmii-id";
+	phy-handle = <&phy1>;
+
+	gmac-config {
+		device = <&gmac>;
+		rgmii-gmac0 = <1>;
+                rxd-delay = <3>;
+                rxdv-delay = <3>;
```
Comment (adschm) [inline target/linux/ath79/dts/qca9558_openmesh_a60.dtsi:159]: broken indent.

File: target/linux/ath79/dts/qca9558_openmesh_a60.dtsi [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+				reg = <0x000000 0x040000>;
+				read-only;
+			};
+
+			partition@40000 {
+				label = "u-boot-env";
+				reg = <0x040000 0x010000>;
+			};
+
+			partition@80000 {
```
Comment (adschm) [inline target/linux/ath79/dts/qca9558_openmesh_a60.dtsi:105]: the offset in the node name does not match the reg property here (and for many other partitions and devices ...)

File: target/linux/ath79/dts/qca9558_openmesh_a60.dtsi (hunk 2 above)
Comment (ecsv) [reply target/linux/ath79/dts/qca9558_openmesh_a60.dtsi]: done

File: target/linux/ath79/dts/qca9558_openmesh_a60.dtsi (hunk 4 above)
Comment (ecsv) [reply target/linux/ath79/dts/qca9558_openmesh_a60.dtsi]: thanks, should be fixed now

File: target/linux/ath79/dts/qca9558_openmesh_a60.dtsi (hunk 3 above)
Comment (ecsv) [reply target/linux/ath79/dts/qca9558_openmesh_a60.dtsi]: thanks, should be fixed now

**Comment by adschm** [comment]
Hi, I will merge the devices from this PR (A60/A40) tonight or tomorrow.

**Pull request closed without merging**